coalib/processes/LogPrinterThread.py

```diff
@@ -19,6 +22,13 @@ def run(self):
         while self.running:
             try:
                 elem = self.message_queue.get(timeout=0.1)
-                self.log_printer.log_message(elem)
+                if self.log_printer is not None and getattr(
+                        self.log_printer, 'log_message'):
```

Comment:
you can do `self.log_printer.log_message` instead of `getattr`, short-circuitance prevents invalid field accesses if `self.log_printer is None`.